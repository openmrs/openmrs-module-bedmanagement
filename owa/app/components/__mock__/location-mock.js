Object.defineProperties(window.location, {
    host : {
        writable: true,
        value: '192.168.33.10'
    },
    hostname : {
        writable: true,
        value: '192.168.33.10'
    },
    href :{
        writable: true,
        value: 'https://192.168.33.10/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    origin :{
        writable: true,
        value: 'https://192.168.33.10'
    },
    pathname :{
        writable: true,
        value: '/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    port :{
        writable: true,
        value: ''
    },
    protocol :{
        writable: true,
        value: 'https:'
    }
});