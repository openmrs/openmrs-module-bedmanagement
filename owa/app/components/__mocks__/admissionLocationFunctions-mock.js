import pickBy from 'lodash.pickby';
import {admissionLocations, visitLocations, bedTypes} from 'utilities/__tests__/testData.json';

const admissionLocationFunctionsMock = {
    setActiveLocationUuid: (admissionLocationUuid) => jest.fn(),
    getActiveLocationUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    setState: (data) => jest.fn(),
    getState: () => jest.fn(),
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    getVisitLocations: () => {
        return visitLocations;
    },
    getBedTypes: () => {
        return bedTypes;
    },
    getAdmissionLocationByUuid: (admissionLocationUuid) => {
        return typeof admissionLocations[admissionLocationUuid] !== 'undefined' ?
            admissionLocations[admissionLocationUuid] : null;
    },
    getParentAdmissionLocation: (admissionLocationUuid) => {
        const parentAdmissionLocationUuid = admissionLocations[admissionLocationUuid].parentAdmissionLocationUuid;
        return parentAdmissionLocationUuid != null ? admissionLocations[parentAdmissionLocationUuid] : null;
    },
    getChildAdmissionLocations : (admissionLocationUuid) => {
        return  pickBy(admissionLocations, (admissionLocation) => {
            return admissionLocation.parentAdmissionLocationUuid == admissionLocationUuid;
        });
    },
    reFetchAllAdmissionLocations : () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default admissionLocationFunctionsMock;
