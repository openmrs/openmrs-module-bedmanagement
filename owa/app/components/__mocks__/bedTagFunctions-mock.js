import _ from 'lodash';

import {bedTags} from 'utilities/__tests__/testData.json';

const bedTagFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTags: () => {
        return bedTags;
    },
    getBedTagByUuid: (bedTagUuid) => {
        return _.find(bedTags, function (bedTag) {
            return bedTag.uuid == bedTagUuid;
        });
    },
    fetchBedTags: () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default bedTagFunctionsMock;
