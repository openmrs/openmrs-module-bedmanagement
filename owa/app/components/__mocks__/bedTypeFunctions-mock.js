import _ from 'lodash';

import {bedTypes} from 'utilities/__tests__/testData.json';

const bedTypeFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTypes: () => {
        return bedTypes;
    },
    getBedTypeById: (bedTypeId) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.id == bedTypeId;
        });
    },
    getBedTypeName: (bedTypeName) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.name == bedTypeName;
        });
    },
    fetchBedTypes: () => {
        return bedTypes;
    },
    notify : (notifyType, message) => jest.fn()
};

export default bedTypeFunctionsMock;
