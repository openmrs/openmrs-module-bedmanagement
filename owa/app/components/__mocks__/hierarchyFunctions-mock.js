import {admissionLocations} from 'utilities/__tests__/testData.json';

const hierarchyFunctionMock = {
    getActiveUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    setActiveUuid: (locationUuid) => jest.fn(),
    setState: (stateData) => jest.fn(),
    toggleIsOpen: (locationUuid, isOpen) => jest.fn(),
    setAdmissionLocationIsOpen: (locationUuid, isOpen) => jest.fn()
};

export default hierarchyFunctionMock;