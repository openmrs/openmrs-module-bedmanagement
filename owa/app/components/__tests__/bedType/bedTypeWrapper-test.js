import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import BedTypeWrapper from 'components/bedType/bedTypeWrapper';
require('components/__mocks__/location-mock');
require('babel-polyfill');
const testProps = {
    match:{
        isExact: true,
        params:{},
        path:'/owa/bedmanagement/bedTypes.html',
        url:'/owa/bedmanagement/bedTypes.html'
    },
    sleep: (milisec) => {
        return new Promise((resolve, reject) => {
            setTimeout(() => {
                return resolve(true);
            }, milisec);
        });
    }
};

describe('BedTypeWrapper', () => {
    beforeAll(() => {
        //jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;

        var mock = new MockAdapter(axios);
        const data = {
            results: [{
                'id': 1,
                'name': 'deluxe',
                'displayName': 'deluxe bed',
                'description': 'DLX'
            },
            {
                'id': 2,
                'name': 'luxury',
                'displayName': 'luxury bed',
                'description': 'LXY'
            },
            {
                'id': 3,
                'name': 'dbl',
                'displayName': 'double bed',
                'description': 'deluxe'
            }]
        };

        mock.onGet('https://192.168.33.10/openmrs/ws/rest/v1/bedtype')
            .reply(200, data);
    });

    it('Should render bed Type page properly', async () =>{
        let bedTypeWrapper = shallow(<BedTypeWrapper match={testProps.match}/>);
        const bedTypeFunctions = bedTypeWrapper.instance().bedTypeFunctions;

        await testProps.sleep(100);
        expect(bedTypeWrapper.find('BedTypeList').length).toBe(1);
        expect(shallowToJson(bedTypeWrapper)).toMatchSnapshot();

        bedTypeFunctions.setState({
            activePage: 'addEdit',
            pageData: {
                operation: 'add',
                bedTypeId: null
            }
        });

        bedTypeWrapper.update();
        expect(bedTypeWrapper.find('AddEditBedType').length).toBe(1);
        expect(shallowToJson(bedTypeWrapper)).toMatchSnapshot();
    });

    it('Should work functions properly', async () =>{
        const bedTypeWrapper = shallow(<BedTypeWrapper match={testProps.match}/>);
        const bedTypeFunctions = bedTypeWrapper.instance().bedTypeFunctions;

        expect(bedTypeFunctions.getBedTypes()).toEqual([]);

        await testProps.sleep(100);
        expect(bedTypeFunctions.getBedTypes()).toEqual([
            { id: 1,
                name: 'deluxe',
                displayName: 'deluxe bed',
                description: 'DLX' },
            { id: 2,
                name: 'luxury',
                displayName: 'luxury bed',
                description: 'LXY' },
            { id: 3,
                name: 'dbl',
                displayName: 'double bed',
                description: 'deluxe' }
        ]);

        expect(bedTypeFunctions.getBedTypeById(2)).toEqual({
            id: 2,
            name: 'luxury',
            displayName: 'luxury bed',
            description: 'LXY'
        });

        expect(bedTypeFunctions.getBedTypeName('deluxe')).toEqual({
            description: 'DLX',
            displayName: 'deluxe bed',
            id: 1,
            name: 'deluxe'
        });


        bedTypeFunctions.setState({
            activePage: 'addEdit',
            pageData: {
                operation: 'add',
                bedTypeId: null
            }
        });

        expect(bedTypeWrapper.state().activePage).toEqual('addEdit');
        expect(bedTypeWrapper.state().pageData).toEqual({
            operation: 'add',
            bedTypeId: null
        });
    });

});