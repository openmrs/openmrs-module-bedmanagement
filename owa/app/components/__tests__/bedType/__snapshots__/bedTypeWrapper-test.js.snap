// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BedTypeWrapper Should render bed Type page properly 1`] = `
<div>
  <Notify />
  <Header
    path="/owa/bedmanagement/bedTypes.html"
  />
  <div
    style={
      Object {
        "marginTop": 30,
      }
    }
  >
    <BedTypeList
      bedTypeFunctions={
        Object {
          "fetchBedTypes": [Function],
          "getBedTypeById": [Function],
          "getBedTypeName": [Function],
          "getBedTypes": [Function],
          "notify": [Function],
          "setState": [Function],
        }
      }
      bedTypes={Array []}
    />
  </div>
</div>
`;

exports[`BedTypeWrapper Should render bed Type page properly 2`] = `
<div>
  <Notify />
  <Header
    path="/owa/bedmanagement/bedTypes.html"
  />
  <div
    style={
      Object {
        "marginTop": 30,
      }
    }
  >
    <AddEditBedType
      bedTypeFunctions={
        Object {
          "fetchBedTypes": [Function],
          "getBedTypeById": [Function],
          "getBedTypeName": [Function],
          "getBedTypes": [Function],
          "notify": [Function],
          "setState": [Function],
        }
      }
      bedTypeId={null}
      operation="add"
    />
  </div>
</div>
`;
